xdsl/dialects/hw.py

```diff
@@ -646,9 +663,18 @@ def __init__(
     def __len__(self) -> int:
         return self.size_attr.data
 
-    def get_element_type(self) -> IntegerType:
+    def get_element_type(self) -> ArrayElementCovT:
         return self.element_type
 
+    @staticmethod
+    def constr(
+        element_type: IRDLAttrConstraint[ArrayElementCovT],
+        size_attr: IRDLAttrConstraint[IntAttr],
+    ) -> AttrConstraint:
+        return ParamAttrConstraint[ArrayType[ArrayElementCovT]](
+            ArrayType, (element_type, size_attr)
+        )
+
```

Comment:
@superlopuh I have the feeling I should be changing, this? Is that possible?
I'm quite stuck. If I try your suggestion to change the bound type to AnySignlessInteger, it does not allow nested types anymore...

Reply:
Ah I see, indeed my suggestion is not correct. I think I would have to play with it to make sure but I don't think I'll have the bandwidth until the end of the week, happy to have a go then if that's not too late for you